### PR TITLE
Support custom rz-pm-db URL

### PR DESCRIPTION
**Pull request by afjoseph**
Hey guys. This PR adds an environment variable to support a custom rz-pm-db.

It also adds a new flag to disable DB updatesa

Finally, I also added a log line to tell the user the package is downloading. This is purely for sanity purposes since I thought multiple times the binary hanged when it was just downloading a large repo.

I've added an explanation for this in the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,14 @@ Download the rz-pm binary for your system on the [latest release page](https://g
 | **GithubCI**  | [![Go](https://github.com/rizinorg/rz-pm/actions/workflows/go.yml/badge.svg)](https://github.com/rizinorg/rz-pm/actions/workflows/go.yml) |
 
 # Available packages
-The official database is available [here](https://github.com/rizinorg/rz-pm-db).
+
+The official database is available [here](https://github.com/rizinorg/rz-pm-db). You can change the repo by supplying a custom `RZPM_DB_REPO_URL` environment flag like this:
+
+    $ RZPM_DB_REPO_URL=https://github.com/bunnyfoofoo/my-custom-rz-pm-db rz-pm install rz-custom-plugin
```
Comment (XVilka) [inline README.md:22]: Please use backticks syntax for code snippets

**Review by XVilka — CHANGES_REQUESTED** [review]
Nice addition, thank you!

File: README.md (hunk 1 above)
Comment (afjoseph) [reply README.md]: Fixed and squased all fixup commits

File: main.go [hunk 2]
```diff
@@ -324,6 +326,11 @@ RZ_PM_SITE:
 			Name:  flagSkipUpgrade,
 			Usage: "skip auto-upgrade on start",
 		},
+		&cli.BoolFlag{
+			Name:  flagShouldUpdateDB,
+			Usage: "should update the DB?",
```
Comment (ret2libc) [inline main.go:331]:
```suggestion
			Usage: "update the DB",
```

**Review by ret2libc — COMMENTED** [review]
Thanks a lot!

File: README.md [hunk 3]
```diff
@@ -16,7 +16,18 @@ Download the rz-pm binary for your system on the [latest release page](https://g
...
+The official database is available [here](https://github.com/rizinorg/rz-pm-db). You can change the repo by supplying a custom `RZPM_DB_REPO_URL` environment flag like this:
+
+```
+$ RZPM_DB_REPO_URL=https://github.com/bunnyfoofoo/my-custom-rz-pm-db rz-pm install rz-custom-plugin
+```
+
+Furthermore, to aid with debugging, you can disable auto-updating the `rz-pm-db` upon each command execution by adding `-should-update-db=false` flag, like this:
+
+```
+$ rz-pm -should-update-db=false install rz-custom-plugin
```
Comment (XVilka) [inline README.md:29]: @afjoseph please change the option to `-update-db` and it would be ready to be merged.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -16,7 +16,18 @@ Download the rz-pm binary for your system on the [latest release page](https://g
...
+```
```
[the remark this answers is not in the document]
Comment (afjoseph) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (afjoseph) [reply README.md]: Done and done

**Comment by afjoseph** [comment]
All changes addressed. Thanks, guys ❤️

File: main.go [hunk 5]
```diff
@@ -324,6 +326,11 @@ RZ_PM_SITE:
 			Name:  flagSkipUpgrade,
 			Usage: "skip auto-upgrade on start",
 		},
+		&cli.BoolFlag{
+			Name:  flagUpdateDB,
+			Usage: "should update the DB?",
```
Comment (XVilka) [inline main.go:331]: You forgot to change this one

**Review by XVilka — CHANGES_REQUESTED** [review]

File: pkg/site.go [hunk 6]
```diff
@@ -64,7 +75,7 @@ const dbDir string = "rz-pm-db"
 const artifactsDir string = "artifacts"
 const installedFile string = "installed"
 
-func InitSite(path string) (Site, error) {
+func InitSite(path string, shouldUpdateDB bool) (Site, error) {
```
Comment (XVilka) [inline pkg/site.go:78]: Rename the variable too

**Review by XVilka — CHANGES_REQUESTED** [review]

File: pkg/site.go (hunk 6 above)
Comment (afjoseph) [reply pkg/site.go]: Done

**Review by ret2libc — APPROVED** [review]

File: pkg/database.go [hunk 7]
```diff
@@ -24,12 +24,14 @@ var ErrRizinPackageWrongHash = errors.New("wrong hash")
 
 const dbPath string = "db"
 
-func InitDatabase(path string, rizinVersion string) (Database, error) {
+func InitDatabase(path string, rizinVersion string, shouldUpdateDB bool) (Database, error) {
```
Comment (XVilka) [inline pkg/database.go:27]: Missed this one.

**Review by XVilka — APPROVED** [review]
Apart from that nitpick LGTM

File: pkg/database.go (hunk 7 above)
Comment (afjoseph) [reply pkg/database.go]: Damn. I thought I got them all. Fixed!

**Comment by ret2libc** [comment]
Thanks a lot!

**Pull request merged**